pkg/distributor/distributor_test.go

```diff
@@ -288,8 +288,18 @@ func TestDistributor_Push(t *testing.T) {
 
 			request := makeWriteRequest(tc.samples.startTimestampMs, tc.samples.num, tc.metadata, false)
 			response, err := ds[0].Push(ctx, request)
-			assert.Equal(t, tc.expectedResponse, response)
-			assert.Equal(t, tc.expectedError, err)
+
+			if tc.expectedError == nil {
+				require.NoError(t, err)
+				assert.Equal(t, emptyResponse, response)
```

Comment:
I just realize that this change makes `tc.expectedResponse` superfluous, in that case we should remove it? It's pointless to keep it around if unused.

Reply:
done in https://github.com/grafana/mimir/pull/3219/commits/0ba6fbd6ada9b78709b250a036a41166007e66b1